elixir/apps/web/lib/web/live/sign_up.ex

```diff
@@ -32,17 +33,43 @@ defmodule Web.SignUp do
         with: fn _account, attrs -> Actors.Actor.Changeset.create(attrs) end
       )
     end
+
+    defp validate_email_allowed(changeset, true, _allowed_domains), do: changeset
+
+    defp validate_email_allowed(changeset, _sign_up_enabled, allowed_domains) do
+      Ecto.Changeset.validate_change(changeset, :email, fn :email, email ->
+        if email_allowed?(email, allowed_domains),
+          do: [],
+          else: [email: "email domain is not allowed at this time"]
+      end)
+    end
+
+    defp email_allowed?(email, allowed_domains) do
+      with [_, domain] <- String.split(email, "@", parts: 2) do
+        Enum.member?(allowed_domains, domain)
+      else
+        _ -> false
+      end
+    end
   end
 
   def mount(_params, _session, socket) do
     user_agent = Phoenix.LiveView.get_connect_info(socket, :user_agent)
     real_ip = Web.Auth.real_ip(socket)
 
     changeset =
-      Registration.changeset(%{
-        account: %{slug: "placeholder"},
-        actor: %{type: :account_admin_user}
-      })
+      Registration.changeset(
+        %{
+          account: %{slug: "placeholder"},
+          actor: %{type: :account_admin_user}
+        },
+        true,
+        []
+      )
+
+    allowed_domains =
+      Domain.Config.get_env(:domain, :sign_up_allowed_domains)
+      |> String.split(",", trim: true)
```

Comment:
The application environment should already contain values that are casted to whatever our application needs (see my previous comment)